### PR TITLE
Add docs for file compression config

### PR DESCRIPTION
**Pull request by mosabua**
## Description

* Necessary since we upgraded to 3.0
* Docs for fallback switch in case there are issues

## Additional context and related issues

Ideally we should get this merged for 456 .. if so I will rebase release notes PR and link to the section.

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

File: docs/src/main/sphinx/admin/properties-general.md [hunk 1]
```diff
@@ -67,3 +67,24 @@ size limits.
...
+Trino uses the [aircompressor](https://github.com/airlift/aircompressor) library
+to compress and decompress ORC, Parquet, and other files using the LZ4, zstd,
+Snappy, and other algorithms. The library takes advantage of using embedded,
+higher performing, native implementations for these algorithms by default. 
+
+If necessary, this behavior can be deactivated to fall back on JVM-based
+implementations with the following configuration:
+
+```properties
+io.airlift.compress.v3.disable-native=true
```
Comment (martint) [inline docs/src/main/sphinx/admin/properties-general.md:83]: This needs to go in the jvm.config file, not the server config properties.

**Review by martint — COMMENTED** [review]
Some of the changes in this commit are unrelated to the new config options. Split them out.

File: docs/src/main/sphinx/admin/properties-general.md (hunk 1 above)
Comment (mosabua) [reply docs/src/main/sphinx/admin/properties-general.md:83]: oh ..

Comment (mosabua) [reply docs/src/main/sphinx/admin/properties-general.md:83]: Updated

File: docs/src/main/sphinx/installation/deployment.md [hunk 2]
```diff
@@ -162,12 +162,18 @@ Because an `OutOfMemoryError` typically leaves the JVM in an
...
+
+The temporary directory used by the JVM must allow execution of code, because
+Trino accesses and uses shared library binaries for purposes such as
+[](file-compression). 
+
+Specifically, the partition mount and directory must not have the `noexec` flag
+set. The default `/tmp` directory is mounted with this flag in some operating
+system installations, which prevents Trino from starting. You can workaround
+this by overriding the temporary directory by adding
+`-Djava.io.tmpdir=/path/to/other/tmpdir` to the list of JVM options.
```
Comment (martint) [inline docs/src/main/sphinx/installation/deployment.md:176]: This should also be an option in jvm.config

File: docs/src/main/sphinx/admin/properties-general.md [hunk 3]
```diff
@@ -67,3 +67,24 @@ size limits.
...
+implementations with the following configuration in the [](jvm-config)
+
+```properties
+-Dio.airlift.compress.v3.disable-native=true
+```
+
+The library relies on the [temporary directory used by the JVM](tmp-directory),
+including the execution of code in the directory, to load the embedded shared
+libraries. If this directory is not suitable you can configure usage of a
+separate directory with an absolute path set with the `aircompressor.tmpdir`
```
Comment (mosabua) [inline docs/src/main/sphinx/admin/properties-general.md:89]: is this also a jvm config? .. i am digging in the code atm

File: docs/src/main/sphinx/admin/properties-general.md (hunk 1 above)
Comment (mosabua) [reply docs/src/main/sphinx/admin/properties-general.md]: fixed

File: docs/src/main/sphinx/installation/deployment.md (hunk 2 above)
Comment (mosabua) [reply docs/src/main/sphinx/installation/deployment.md:176]: fixed

File: docs/src/main/sphinx/admin/properties-general.md [hunk 4]
```diff
@@ -67,3 +67,28 @@ size limits.
...
+If necessary, this behavior can be deactivated to fall back on JVM-based
+implementations with the following configuration in the [](jvm-config):
+
+```properties
+-Dio.airlift.compress.v3.disable-native=true
+```
+
+The library relies on the [temporary directory used by the JVM](tmp-directory),
+including the execution of code in the directory, to load the embedded shared
+libraries. If this directory is not suitable you can configure usage of a
```
Comment (wendigo) [inline docs/src/main/sphinx/admin/properties-general.md:88]: I'd call the `noexec` mount option directly

File: docs/src/main/sphinx/admin/properties-general.md (hunk 3 above)
Comment (martint) [reply docs/src/main/sphinx/admin/properties-general.md]: Yes

File: docs/src/main/sphinx/admin/properties-general.md (hunk 4 above)
Comment (wendigo) [reply docs/src/main/sphinx/admin/properties-general.md:88]: nvm, it's lower :)

**Review by wendigo — APPROVED** [review]

**Comment by mosabua** [comment]
> Some of the changes in this commit are unrelated to the new config options. Split them out.

Its really all interdependent. I am not sure how to pull the commits apart in a very meaningful way since its all interrelated.

File: docs/src/main/sphinx/admin/properties-general.md (hunk 4 above)
Comment (mosabua) [reply docs/src/main/sphinx/admin/properties-general.md]: Done

**Comment by martint** [comment]
How is this related to the airlift config docs, for instance?

> [](file-compression) is automatically performed and some details can
be configured.

**Comment by mosabua** [comment]
> How is this related to the airlift config docs, for instance?
> 
> > `[](file-compression) is automatically performed and some details can be configured.`

well.. it links to the new compression stuff. I guess I can pull it into a follow up commit .. let me think about how to untangle this best

**Review by martint — APPROVED** [review]

**Pull request merged**